cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpReservedIpResourceBuilder.java

```diff
@@ -61,18 +80,23 @@
             try {
                 Operation operation = networkInsert.execute();
                 if (operation.getHttpErrorStatusCode() != null) {
-                    throw new GcpResourceException(operation.getHttpErrorMessage(), resourceType(), resource.getName());
+                    throw new GcpResourceException(operation.getHttpErrorMessage(), ResourceType.GCP_RESERVED_IP, resource.getName());
                 }
-                result = Collections.singletonList(createOperationAwareCloudResource(resource, operation));
+                result.add(createOperationAwareCloudResource(resource, operation));
             } catch (GoogleJsonResponseException e) {
-                throw new GcpResourceException(checkException(e), resourceType(), resource.getName());
+                throw new GcpResourceException(e.getDetails().getMessage(), ResourceType.GCP_RESERVED_IP, resource.getName());
```

Comment:
Why did you change this line from using methods to pulling the logic out and hardcoding it? I know the methods are very simple, but why have this code more than one place when you can just call the methods available via the abstract classes?

Reply:
originally changed to make it a static method. not needed with injection